src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -758,10 +759,41 @@ private static bool IsAllPunctuation(string filterText)
         /// </summary>
         private static bool IsPotentialFilterCharacter(char c)
         {
-            // TODO(cyrusn): Actually use the right unicode categories here.
+            // TODO(cyrusn): Actually use the right Unicode categories here.
             return char.IsLetter(c)
                 || char.IsNumber(c)
                 || c == '_';
         }
+
+        /// <summary>
+        /// This comparer can handle expanded items.
+        /// </summary>
+        private class CompletionItemSortingComparer : IComparer<VSCompletionItem>
+        {
+            public static CompletionItemSortingComparer Instance { get; } = new CompletionItemSortingComparer();
+
+            public int Compare(VSCompletionItem x, VSCompletionItem y)
+            {
+                var xHasPrefix = HasExpandedPrefix(x);
+                var yHasPrefix = HasExpandedPrefix(y);
+
+                if (xHasPrefix == yHasPrefix)
+                {
+                    // Don't use Ordinal because we want lowercase to be "smaller" than uppercase.
+                    // e.g. "Goo" before "GoTo"
+                    return string.Compare(x.SortText, y.SortText, StringComparison.InvariantCulture);
+                }
+                else if (xHasPrefix)
+                {
+                    return 1;
+                }
+                else
+                {
+                    return -1;
+                }
+
+                static bool HasExpandedPrefix(VSCompletionItem i) => i.SortText.Length > 1 && i.SortText[0] == ImportCompletionItem.SortTextPrefix;
```

Comment:
a local function here is overkill for me.  this is already a self contained class for literally just this purpose.  can we just make this a clear static helper method?